Automattic/wp-calypso
# chore: Extract `babel-plugin-optimize-i18n` to a separate package

#### Changes proposed in this Pull Request

Extracts babel-plugin-optimize-i18n ([original PR](https://github.com/Automattic/wp-calypso/pull/49341)) to a separate package `@automattic/babel-plugin-optimize-i18n`.

The main reason is I'm trying to split `@automattic/calypso-build` into smaller, granular parts that don't require users to pull in all peer dependencies.

#### Testing instructions

Run `rm -fr .cache build public && NODE_ENV=production CALYSPO_ENV=production BUILD_TRANSLATION_CHUNKS=true yarn build-client` in this branch and `trunk` and verify the output is the same (`diff -qr wp-calypso/public/evergreen wp-calypso2/public/evergreen`)
